### PR TITLE
treewide: remove deprecations up until 24.11

### PR DESCRIPTION
**Pull request by wolfgangwalther**
Grepping through the repo for NixOS version numbers brought up those candidates for removal. Many were targeted at 24.11, but some have already been forgotten for a while.

If any of those are controversial, I can split those into a separate PR.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Aleksanaa** [comment]
My suggestion is not to backport to 24.11. It has feature frozen. Considering we won't be making major changes to it, we won't lose anything by not backporting it, after all

**Review by hsjobeki — CHANGES_REQUESTED** [review]

File: doc/redirects.json
Comment (GetPsyched) [reply doc/redirects.json:1422]: Sorry for the inconvenience caused. A helper command to update the redirects is in the works, which should make most [tm] of these updates semi-automatic.

**Comment by emilazy** [comment]
I think that if we don’t merge this for 24.11 we ought to at least update the error messages to say 25.05, which seems worse than just getting it done with. There seemed to be consensus in the release management room on Matrix that removing obsolete/insecure stuff is okay up until release as long as it’s not adding additional risk for Nixpkgs itself. I’m personally inclined to merge + backport this; the release has been delayed another week because of having to restart the `staging-next-24.11` cycle anyway.

**Review by wegank — APPROVED** [review]

**Review by emilazy — APPROVED** [review]
Approved by release manager; ping me after ofborg eval is done for merge.

**Comment by emilazy** [comment]
BTW, it looks like this may leave some `({ … })` that could be cleaned up (but this is non‐blocking).

**Review by GetPsyched — APPROVED** [review]

**Comment by emilazy** [comment]
Looks good, thanks!

**Review by szlend — APPROVED** [review]
LGTM for redocly. I should have done this myself. Though personally I would just keep it as is since we're already past the change freeze.

**Comment by emilazy** [comment]
How ironic.

**Pull request merged**
